dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala

```diff
@@ -0,0 +1,306 @@
+  ECPrivateKey,
+  ECPublicKey,
+  Schnorr,
+  SchnorrDigitalSignature
+}
+import org.bitcoins.core.currency.CurrencyUnit
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.script.{
+  CLTVScriptPubKey,
+  ConditionalScriptPubKey,
+  MultiSignatureScriptPubKey,
+  MultiSignatureWithTimeoutScriptPubKey,
+  P2PKHScriptPubKey,
+  ScriptPubKey
+}
+import org.bitcoins.core.protocol.transaction.{
+  Transaction,
+  TransactionOutPoint,
+  TransactionOutput
+}
+import org.bitcoins.core.script.constant.ScriptNumber
+import org.bitcoins.core.script.crypto.HashType
+import org.bitcoins.core.util.CryptoUtil
+import org.bitcoins.core.wallet.builder.BitcoinTxBuilder
+import org.bitcoins.core.wallet.fee.FeeUnit
+import org.bitcoins.core.wallet.utxo.{
+  BitcoinUTXOSpendingInfo,
+  ConditionalPath,
+  ConditionalSpendingInfo,
+  MultiSignatureSpendingInfo
+}
+import scodec.bits.ByteVector
+
+import scala.concurrent.{ExecutionContext, Future}
+
+// Two selfs: Local, Remote
+// Two outcomes: Win, Lose
+case class BinaryOutcomeDLCWithSelf(
+    outcomeWin: String,
+    outcomeLose: String,
+    oraclePubKey: ECPublicKey,
+    preCommittedR: ECPublicKey,
+    fundingLocalPrivKey: ECPrivateKey,
+    fundingRemotePrivKey: ECPrivateKey,
+    cetLocalPrivKey: ECPrivateKey,
+    cetRemotePrivKey: ECPrivateKey,
+    finalLocalPrivKey: ECPrivateKey,
+    finalRemotePrivKey: ECPrivateKey,
+    localInput: CurrencyUnit,
+    remoteInput: CurrencyUnit,
+    localFundingUtxos: Vector[BitcoinUTXOSpendingInfo],
+    remoteFundingUtxos: Vector[BitcoinUTXOSpendingInfo],
+    localWinPayout: CurrencyUnit,
+    remoteWinPayout: CurrencyUnit,
+    localLosePayout: CurrencyUnit,
+    remoteLosePayout: CurrencyUnit,
+    timeout: Int,
+    feeRate: FeeUnit,
+    changeSPK: ScriptPubKey,
+    network: BitcoinNetwork)(implicit ec: ExecutionContext) {
+
+  val messageWin: ByteVector =
+    CryptoUtil.sha256(ByteVector(outcomeWin.getBytes)).flip.bytes
+
+  val messageLose: ByteVector =
+    CryptoUtil.sha256(ByteVector(outcomeLose.getBytes)).flip.bytes
+
+  val sigPubKeyWin: ECPublicKey =
+    Schnorr.computePubKey(messageWin, preCommittedR, oraclePubKey)
+
+  val sigPubKeyLose: ECPublicKey =
+    Schnorr.computePubKey(messageLose, preCommittedR, oraclePubKey)
+
+  private val totalInput = localInput + remoteInput
+  private val fundingUtxos = localFundingUtxos ++ remoteFundingUtxos
+
+  val fundingLocalPubKey: ECPublicKey = fundingLocalPrivKey.publicKey
+  val fundingRemotePubKey: ECPublicKey = fundingRemotePrivKey.publicKey
+
+  val fundingSPK: MultiSignatureScriptPubKey = {
+    MultiSignatureScriptPubKey(2,
+                               Vector(fundingLocalPubKey, fundingRemotePubKey))
+  }
+
+  /** Subtracts the estimated fee from the last output's value */
+  private def subtractFeeAndSign(
+      txBuilder: BitcoinTxBuilder): Future[Transaction] = {
+    txBuilder.unsignedTx.flatMap { tx =>
+      val fee = feeRate.calc(tx)
```

Comment:
So i think i might be missing something here

Why doesn't this "dummy signing" process work for this script? 

https://github.com/bitcoin-s/bitcoin-s/blob/master/core/src/main/scala/org/bitcoins/core/wallet/builder/TxBuilder.scala#L174

You can see that the flag there is `true`, so that we should place "dummy" signatures where real ones should go. Everything else should be the same as this signing dry run expect when we call the real `.sign()` method, those dummy signatures will be replaced with real ones.

There is a problem of digital signatures varying in size -- IIRC they can vary between 68 bytes and 73 bytes -- but I don't think that should matter for a simple 2 of 2 multisig or a timeout clause.